backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/impl/ElasticsearchSearchScopeModel.java

```diff
@@ -109,14 +111,33 @@ else if ( !componentRetrievalStrategy.areCompatible( selectedComponent, componen
 							)
 					);
 				}
+				else if ( !componentRetrievalStrategy.hasCompatibleConverter( selectedComponent, component ) ) {
+					dslConverterHandler = new ElasticsearchIncompatibleDslConverterHandler<>(
+							absoluteFieldPath, selectedComponent, component, EventContexts.fromIndexNames(
+							indexModelForSelectedSchemaNode.getHibernateSearchIndexName(),
+							indexModel.getHibernateSearchIndexName()
+					), componentRetrievalStrategy );
+				}
 			}
 		}
 		if ( selectedSchemaNode == null ) {
 			throw log.unknownFieldForSearch( absoluteFieldPath, getIndexesEventContext() );
 		}
+		if ( dslConverterHandler == null ) {
+			// no converter incompatibility detected
+			dslConverterHandler = new ElasticsearchDslConverterHandler();
+		}
+
 		return selectedComponent;
 	}
 
+	public ElasticsearchDslConverterHandler getDslConverterHandler() {
+		if ( dslConverterHandler == null ) {
+			throw new IllegalStateException( "This method is supposed to be called after #getSchemaNodeComponent()" );
```

Comment:
I really, really don't like this. Expecting the client to call methods in the right order... it's terrible practice.